create unified BE path for isomorphic visualizations

alxnddr opened the pull request:
### Description

First static visualizations that used javascript for rendering were incomplete duplicates which tried to reproduce the logic of the dynamic visualizations. The logic was spread between Clojure and Javascript. This approach has multiple downsides because all viz features have to be implemented twice and the static chart versions were far behind.

With the new [technical vision](https://www.notion.so/metabase/Viz-Revamp-Research-and-Planning-4198b2726ac64f1d89ad1de6187532d2) for visualizations we will be using the same code both for dynamic and static charts. This PR creates a single BE rendering path for isomorphic visualizations and removes code of the BE code of line/area/bar/combo static charts (FE will be removed further).

### How to verify

1. Create a dashboard with line/area/bar/combo charts and two combined scalars on a single card.
2. Send a test subscription
3. Ensure the placeholders are being rendered

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

alxnddr commented:
Current dependencies on/for this PR:
* master
  * **PR #34606** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34606" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
    * **PR #34717** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34717" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
      * **PR #34772** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34772" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
        * **PR #34809** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34809" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
          * **PR #34816** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34816" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
    * **PR #34703** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34703" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
      * **PR #34763** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34763" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 
        * **PR #34820** <a href="https://app.graphite.dev/github/pr/metabase/metabase/34820" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/metabase/metabase/34820?utm_source=stack-comment).

alxnddr committed:
remove old line/area/bar/combo charts paths

alxnddr committed:
include dashcard viz settings into default computation

EmmadUsmani committed:
fix `getCommonStaticVizSettings`

markbastian commented on test/metabase/pulse/render/body_test.clj:
```diff
@@ -404,86 +404,6 @@
 (defn has-inline-image? [rendered]
   (some #{:img} (flatten-html-data rendered)))
 
-(defn- render-bar-graph [results]
```
Is there a reason all of these tests were just removed? Seems like they should be ported over to the new system unless they aren't valid any more.

markbastian commented:
I noticed that the following form from `dev.render-png` no longer works.

```clojure
  (-> [["A" "B"]
       [1 2]
       [30 20]]
      (render.tu/make-viz-data :line {:goal-line {:graph.goal_label "Target"
                                                  :graph.goal_value 20}})
      :viz-tree
      open-hiccup-as-html)
```

This results in:

```
Execution error (PolyglotException) at <js>/:program (Unnamed:1).
ReferenceError: combo_chart is not defined
```

markbastian commented:
> removes code of the BE code of line/area/bar/combo static charts (FE will be removed further)

Are we ok with this approach or should we have individual PRs for each chart type where we both remove and replace in a single PR?

markbastian commented:
This looks like the correct verification.

![verify](https://github.com/metabase/metabase/assets/8381441/00725482-4f48-4325-aa7f-44523ee179e9)

I'm still concerned that we're wholesale removing something vs. piecemeal replacing.

adam-james-v committed:
Removing some more failing tests.

These tests are tracked and will be re-added appropriately as the isomorphic chart development continues.

markbastian committed:
Isomorphic timeline events viz 34924 (#35619)

* Initial states of isomorphic line charts/events for BE

Initial stages of getting line charts, with a focus towards timeline events,
rendered using the isomorphic pipeline. This change handles cases where a
card has multiple breakouts, forming multiple timelines, vs. a dashcard, which
may have both a query and series to establish multiple timelines.

* Timeline events passed to :isomorphic chart type

Timeline events are now grabbed for the card(s) and passed into the static-viz rendering pipeline.

This commit adds:

 - `dashcard-timeline-events` helper fn that gets events for every timeline in the Collection that the card is part
 of.
 - adjustments to the `execute-multi-card` function for the situation where it's called with `nil`
 dashcard-or-id.
   - This function may need to be re-visited as the isomorphic chart work continues (namely, we need to make sure
 this function works correctly for all scenarios 1. single card 2. single card with multiple series (a 'combo' chart)
 3. dashcard with multiple single series visualized on one card and 4. dashcard with mixed single/multi-series
 cards (not sure if this last one is possible).
 - a test to ensure timeline events are added

---------

Co-authored-by: Adam James <adam.vermeer2@gmail.com>
